crates/nu-cmd-extra/src/extra/bits/rotate_left.rs

```diff
@@ -96,61 +111,93 @@ impl Command for BitsRol {
                     Span::test_data(),
                 )),
             },
+            Example {
+                description: "rotate left binary data",
+                example: "0x[c0 ff ee] | bits rol 10",
+                result: Some(Value::binary(vec![0xff, 0xbb, 0x03], Span::test_data())),
+            },
         ]
     }
 }
 
-fn get_rotate_left<T: Display + PrimInt>(val: T, bits: u32, span: Span) -> Value
-where
-    i64: std::convert::TryFrom<T>,
-{
-    let rotate_result = i64::try_from(val.rotate_left(bits));
-    match rotate_result {
-        Ok(val) => Value::int(val, span),
-        Err(_) => Value::error(
-            ShellError::GenericError {
-                error: "Rotate left result beyond the range of 64 bit signed number".into(),
-                msg: format!(
-                    "{val} of the specified number of bytes rotate left {bits} bits exceed limit"
-                ),
-                span: Some(span),
-                help: None,
-                inner: vec![],
-            },
-            span,
-        ),
-    }
-}
+fn action(input: &Value, args: &Arguments, span: Span) -> Value {
+    let Arguments {
+        signed,
+        number_size,
+        bits,
+    } = *args;
 
-fn operate(value: Value, bits: usize, head: Span, signed: bool, number_size: NumberBytes) -> Value {
-    let span = value.span();
-    match value {
+    match input {
         Value::Int { val, .. } => {
             use InputNumType::*;
-            // let bits = (((bits % 64) + 64) % 64) as u32;
-            let bits = bits as u32;
-            let input_type = get_input_num_type(val, signed, number_size);
-            match input_type {
-                One => get_rotate_left(val as u8, bits, span),
-                Two => get_rotate_left(val as u16, bits, span),
-                Four => get_rotate_left(val as u32, bits, span),
-                Eight => get_rotate_left(val as u64, bits, span),
-                SignedOne => get_rotate_left(val as i8, bits, span),
-                SignedTwo => get_rotate_left(val as i16, bits, span),
-                SignedFour => get_rotate_left(val as i32, bits, span),
-                SignedEight => get_rotate_left(val, bits, span),
+            let val = *val;
+            let bits = bits as u64;
+            let input_num_type = get_input_num_type(val, signed, number_size);
+            match input_num_type {
+                One if bits <= 0xff => {
```

Comment:
I'm a bit unclear on these guards. Is rotating single byte integers more than 2^255 something that isn't useful/doesn't make sense?

Reply:
the original implementation uses `i64::try_from`, and errors when trying to rotate an 8-byte int into the negative range (e.g. `1 | bits ror 1 -n 8`).

I think this was supposed to be checking if `val` (instead of `bits`) was in range for the specified int size so that the error could be moved up into the match statement, but that isn't what's being checked and shouldn't be necessary since it's already handled by `get_input_num_type`.
just fixed to put this back in line with the original behavior

fwiw, I am not sure it is necessary to error when rotating i64s into the signed range, but do not know what the implications of changing this default are

Reply:
I see. I'm not sure what the implications to not having it error are either, but changes to existing behavior should be avoided. Also, I do think errorring without `--signed` is reasonable, in case someone is dealing with naturals and doesn't want to get any negative numbers back. That said, you can ask about it in Discord and maybe see if people are fine with it first.